src/bin/cargo/cli.rs

```diff
@@ -408,22 +408,24 @@ fn cli() -> App {
         "cargo [OPTIONS] [SUBCOMMAND]"
     };
     App::new("cargo")
-        .settings(&[
-            AppSettings::UnifiedHelpMessage,
-            AppSettings::DeriveDisplayOrder,
-            AppSettings::VersionlessSubcommands,
```

Comment:
`VersionlessSubcommands` isn't needed, if a subcommand has no version set then it won't have a `--version`